### PR TITLE
Create room as soon as the channel is created

### PR DESCRIPTION
**Pull request by rakanalh**
## Description

Currently, the room is created as soon as the first message is about to be exchanged between participants of a channel (this message is a payment with a lock). However, certain issues in the transport might end up taking longer than usual and that time would be lost from the time at which a transfer can be unlocked. This change optimizes this by creating the room as soon as the channel is created.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [ ] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [ ] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

**Comment by andrevmatos** [comment]
Creating channels earlier can be problematic, if the partner didn't whitelisted us yet. That is the reason why it was done lazily, both for [channel] partners & targets/initiators. Moving it early, at channel creation time or whitelisting time, needs to ensure invites are retried, in case it was received before peer whitelisted us.

File: raiden/network/transport/matrix/transport.py [hunk 1]
```diff
@@ -709,12 +709,6 @@ def _handle_invite(self, room_id: _RoomID, state: dict) -> None:
             )
             return
 
-        if not self._address_mgr.is_address_known(peer_address):
```
[the remark this answers is not in the document]
Comment (rakanalh) [reply raiden/network/transport/matrix/transport.py]: @hackaugusto and I discussed this and there's already a check that the sender has to be whitelisted before we accept messages from it.

**Comment by ulope** [comment]
The spam protection can be re-added in a later PR: #5308

File: raiden/network/transport/matrix/utils.py [hunk 2]
```diff
@@ -246,7 +246,7 @@ def populate_userids_for_address(self, address: Address, force: bool = False) ->
                 ),
             )
 
-    def refresh_address_presence(self, address: Address) -> None:
+    def track_address_presence(self, address: Address, user_ids: Set[str]) -> None:
```
[the remark this answers is not in the document]
Comment (hackaugusto) [reply raiden/network/transport/matrix/utils.py]: @palango @karlb just a heads up

**Comment by hackaugusto** [comment]
- Only the node with the lower address will create rooms. This removes room
  creation races, where both partner nodes try to create a room at the same
  time.
- Only the address with the lower address will invite users to the existing
  rooms. This removes races with node roaming Matrix servers and room
  inviting/joining.
- The address with the higher address just waits to be invited. This is a
  corollary of 1.
- Invites are automatically processed, regardless of the whitelisting. This
  fixes a race condition where the node with the lower address invites the
  partner before it processes the channel open event (IOW, when the partner
  process an invite before the whitelisting is done).
- Only messages of whitelisted nodes are processed. To avoid spamming attacks,
  previously done by ignoring invite of non-whitelisted nodes.
- Rooms are only created during the whitelist. This removes internal race
  conditions for duplicate room creation. (Whitelist can be called twice at the
  same time).
- Only one room _per partner_ is created at any time (IOW, synchronization was
  added). This removes internal race conditions for duplicate room creation.
- The user id presence change callback is now properly called (previously a
  code path would skip it). This is a bug fix.
- The long running presence update greenlet was removed, /sync is the only
  source of presence updates.
- We also added logs, documentation, simplified the exception handling,
  improved the naming of methods and variables.

**Pull request merged**